src/jit/optimizer.cpp

```diff
@@ -7845,12 +7844,9 @@ GenTree *           Compiler::optIsBoolCond(GenTree *   condBranch,
     {
         isBool = true;
     }
-    else if (opr1->gtOper == GT_CNS_INT)
+    else if (opr1->IsZero() || opr1->IsIOne())
```

Comment:
This would permit long and floating const zero, which wasn't permitted earlier. Can it lead to correctness issues due to liberal condition being checked here?